packages/puppeteer-core/src/common/Frame.ts

```diff
@@ -1083,6 +1087,45 @@ export class Frame {
     return this.worlds[PUPPETEER_WORLD].title();
   }
 
+  /**
+   * @internal
+   */
+  _deviceRequestPromptManager(): DeviceRequestPromptManager {
+    if (this.isOOPFrame()) {
+      return this._frameManager._deviceRequestPromptManager(this.#client);
+    }
+    return this.parentFrame()!._deviceRequestPromptManager();
```

Comment:
let's remove `!` here: I guess the main frame does not have a parent, so we need to check for that?

Reply:
Good point. I had been thinking if we `isOOPFrame` is false there must be a `parentFrame` but I guess that is ovely dependent on the current implementation of `isOOPFrame`.